grantPermission.ps1

```diff
@@ -0,0 +1,274 @@
+
+        Write-Verbose 'Adding Authorization headers'
+        $headers = [System.Collections.Generic.Dictionary[[String], [String]]]::new()
+        [void]$headers.Add('Authorization', "$($AccessToken.token_type) $($AccessToken.access_token)")
+        [void]$headers.Add('Accept', 'application/json')
+        [void]$headers.Add('Content-Type', 'application/json')
+        Write-Output $headers
+    }
+    catch {
+        throw $_
+    }
+}
+#endregion functions
+
+try {
+    # Create authorization headers
+    try {
+        $headers = New-AuthorizationHeaders -ClientId $clientId -ClientSecret $clientSecret
+    }
+    catch {
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
+
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error creating authorization headers. Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $true
+            })
+
+        # Throw terminal error
+        throw
+    }
+
+    # Get current account
+    try {
+        Write-Verbose "Querying account with id [$($actionContext.References.Account.Id)]"
+        $splatWebRequest = @{
+            Uri             = "$baseUrl/scim/users/$($actionContext.References.Account.Id)"
+            Headers         = $headers
+            Method          = 'GET'
+            ContentType     = "application/json;charset=utf-8"
+            UseBasicParsing = $true
+        }
+
+        $currentAccount = Invoke-RestMethod @splatWebRequest -Verbose:$false
+
+        if (($currentAccount | Measure-Object).count -gt 1) {
+            throw "Multiple accounts found with id [$($actionContext.References.Account.Id)]. Please correct this so the accounts are unique."
+        }
+        elseif (($currentAccount | Measure-Object).count -eq 0) {
+            throw "No account found with id [$($actionContext.References.Account.Id)]."
+        }
+    }
+    catch {
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
+
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+        Write-Verbose "URI: $($splatWebRequest.Uri)"
+
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error querying account with id [$($actionContext.References.Account.Id)]. Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $true
+            })
+
+        # Throw terminal error
+        throw
+    }
+
+    # Grant permission
+    try {
+        Write-Verbose "Granting permission to $($pRef.Name) ($($pRef.id)) for $($currentAccount.userName) ($($currentAccount.id))"
```

Comment:
the $pRef should be available in the $actionContext in ps v2? This applies to all the $pRef references in the code to follow

Reply:
Correct, changed accordingly.